doc/concepts/pipeline-concepts/pipeline/index.rst

```diff
@@ -0,0 +1,61 @@
+Pipeline
+========
+
+A pipeline is a Pachyderm primitive that is responsible for collecting data
```

Comment:
Do you think we want a different section on DAGs specifically? Or is it enough to just casually mention DAG here and that's probably good enough info for now?

For example, we do get questions about how Pachyderm DAG pipelines are different than airflow so we could maybe have a sentence or two here that talks about how pipelines (and DAGs) in Pachyderm are declarative -- since that's one of the defining characteristics that make our pipeline system unique. Alternatively, we might just need a doc or blog post about "what makes pachyderm pipeline unique compared to most other DAG schedulers:

- Declarative
- Incrementality (e.g. understand versioning)
- Automatically manage provenance
- crazy simple, yet opinionated, data interface: pfs/in and pfs/out
- enable distributed computation on single-threaded code via data sharding

As I'm writing this, it's abundantly obvious that this is a whole doc/blog post of its own. So I kinda answer my own question. I guess this should become its own doc issue but also note that it could be a blog post instead.